<?xml version="1.0"?>
<robot xmlns:xacro="http://wiki.ros.org/xacro" name="$(arg name)">

   <!-- Robot name parameter -->
   <xacro:arg name="name" default="mycobot_320"/>

   <!-- Import main macro -->
   <xacro:include filename="$(find mycobot_320_description)/urdf/mycobot_320_macro.xacro"/>

   <!-- Parameters -->
   <xacro:arg name="prefix" default="" />
   <xacro:arg name="joint_limit_params" default=""/>
   <xacro:arg name="kinematics_params" default=""/>
   <xacro:arg name="physical_params" default=""/>
   <xacro:arg name="visual_params" default=""/>

    <!-- Simulation parameters -->
   <xacro:arg name="use_fake_hardware" default="true" />

   <!-- Initial position for fake hardware -->
   <xacro:arg name="initial_positions_file" default="$(find mycobot_320_description)/config/initial_positions.yaml"/>

   <!-- Create link fixed to the "world" -->
   <link name="world" />

   <!-- Arm -->
   <xacro:mycobot_320_robot
     name="$(arg name)"
     prefix="$(arg prefix)"
     parent="world"
     joint_limits_parameters_file="$(arg joint_limit_params)"
     kinematics_parameters_file="$(arg kinematics_params)"
     physical_parameters_file="$(arg physical_params)"
     visual_parameters_file="$(arg visual_params)"
     initial_positions_file="$(arg initial_positions_file)"
     use_fake_hardware="$(arg use_fake_hardware)">
     <origin xyz="0 0 0" rpy="0 0 0" />          <!-- position robot in the world -->
   </xacro:mycobot_320_robot>


   <!-- Gripper -->
   <xacro:include filename="$(find cobot_demo)/urdf/gripper_macro.xacro"/>
   <xacro:gripper></xacro:gripper>


   <xacro:include filename="$(find cobot_demo)/urdf/gripper.ros2_control.xacro"/>
   <xacro:gripper_ros2_control use_fake_hardware="$(arg use_fake_hardware)">
   </xacro:gripper_ros2_control>



    <!-- Gripper -->
    <!-- <link name="gripper"/> -->
    <joint name="link6-gripper" type="fixed">
      <origin xyz="0 0 0.025" rpy="0 0 0"/>
      <parent link="link6"/>
      <child link="gripper_base"/>
    </joint>

    <!-- Camera frame -->
    <link name="camera"/>
    <joint name="link6-camera" type="fixed">
      <origin xyz="0 0 0.075" rpy="0 0 0"/>
      <parent link="link6"/>
      <child link="camera"/>
    </joint>


    <!-- Target -->
    <link name="target"/>
    <joint name="base-target" type="fixed">
      <origin xyz="-0.035 -0.214 0.353" rpy="${radians(-106.984)} ${radians(-23.960)} ${radians(-172.926)}"/>
      <parent link="base_link"/>
      <child link="target"/>
    </joint>

    <!-- Include gripper -->
    <xacro:include filename="$(find mycobot_320_description)/urdf/inc/mycobot_common.xacro" />







</robot>
